### ISSUE TITLE
No training configuration found

### ISSUE THREAD

**Issue by ChienLiu** [fizyr/keras-retinanet#207]
First of all, I'd like to thank several staff for your efforts to complete such a wonderful project.
Then I'm sorry for the bother caused by my question.

The problem is as follows.
I tried to train a model with VOC by using train.py and 50 epochs, and then I got the final model 'resnet50_pascal_50.h5'.
Next, I loaded the 'resnet50_pascal_50.h5' by the code 
' model = keras.models.load_model('snapshots/resnet50_pascal_50.h5', custom_objects=custom_objects) '

But I got a message

**/usr/local/lib/python2.7/dist-packages/keras/models.py:252: UserWarning: No training configuration found in save file: the model was *not* compiled. Compile it manually.
  warnings.warn('No training configuration found in save file: '**

Was there something I did in wrong way?
Thank you for your reply.

**Comment by huangcong77** [comment]
I think it doesn't matter.Just ignore it. You can still use this model to test and so on.

**Comment by hgaiser** [comment]
Seems you saved it before running `compile` on your model. What the warning means is that no training state is found in the saved model, meaning if you wanted to, you can't resume training. Based on the error I would guess it should still run the detections on your model.

Does this resolve your issue?

**Comment by ChienLiu** [comment]
Thank you very much for the reply.
As huangcong77 said, ignoring this issue seems to have no effect.
At that time, I was a little over-worried about where I did something wrong.
But nothing happened, everything looked very successful.
You guys are very enthusiastic, thanks again.

**Comment by hgaiser** [comment]
Good to hear :)

**Comment by tonmoyborah** [comment]
I seem to find this warning too when I use a saved model to predict. During training I do this in order
create model -> compile model -> data flow from dir -> fit generator -> save model. Then I use the model in another script as load model -> predict.
Is it okay to ignore this , or is it required to compile the model again? I am using Keras 2.2

**Comment by IsaacPatole** [comment]
@hgaiser @ChienLiu @huangcong77 I am also getting the same error while doing inferencing warnings.warn('No training configuration found in save file

Shall I take the last trained model from snapshot, and load it and then compile it like

model = models.load(model.h5)

model.compile(
    loss={
        'regression'    : keras_retinanet.losses.smooth_l1(),
        'classification': keras_retinanet.losses.focal()
    },
    optimizer=keras.optimizers.adam(lr=1e-5, clipnorm=0.001)
)


then again do, model.save('model.h5') ---------> create inference model out of it and then run the inference?

**Comment by hgaiser** [comment]
It's okay to ignore.
